### PR TITLE
ENUM_VALUE styles support: snakeCase/pascalCase

### PR DESCRIPTION
**Pull request by PhilipDukhov**
## Which rule and warnings did you add?

I've added config value for ENUM_VALUE.
Not everyone prefer snake case: https://discuss.kotlinlang.org/t/enum-constants-uppercase-or-camecase/13984
So there should be a possibility to choose what you prefer

## Actions checklist
* [x] Implemented Rule, added Warnings
* [x] Added tests on checks
* [x] Added tests on fixers
* [x] Updated diktat-analysis.yml
* [x] Updated available-rules.md

## Fixme
Not sure how can I use different warning for each case

### CODE REVIEW

**Comment by akuleshov7** [comment]
Yeah, agree with you, it should be configurable. But are you really using enums not in UPPER_SNAKE_CASE?

File: diktat-analysis.yml [hunk 1]
```diff
@@ -10,7 +10,8 @@
   configuration: {}
 - name: ENUM_VALUE
   enabled: true
-  configuration: {}
+  configuration:
+    enumStyle: snakeCase
```
Comment (akuleshov7) [inline diktat-analysis.yml:14]: I am afraid to ask, but may be we need to support all the cases?

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt [hunk 2]
```diff
@@ -378,4 +390,14 @@ class IdentifierNaming(private val configRules: List<RulesConfig>) : Rule("ident
         val prevCatchKeyWord = parentValueParamList?.prevCodeSibling()?.elementType == CATCH_KEYWORD
         return node.text == "e" && node.findParentNodeWithSpecificType(CATCH) != null && prevCatchKeyWord
     }
+
+    enum class EnumStyles {
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:394]: May be rename it to "Style" and move it to StringCaseUtils?

**Review by akuleshov7 — APPROVED** [review]
lgtm

**Comment by PhilipDukhov** [comment]
> Awesome, thanks!!! <3
> Yeah, agree with you, it should be configurable. But are you really using enums not in UPPER_SNAKE_CASE?

Well, I came to Android development from iOS, and UPPER_SNAKE_CASE for me looks crazy 😁

File: diktat-analysis.yml (hunk 1 above)
Comment (PhilipDukhov) [reply diktat-analysis.yml:14]: I know only about usage of these two :)

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt (hunk 2 above)
Comment (PhilipDukhov) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:394]: sure, but for enums I think we only need to support these two

Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:394]: yes, that is fine, let's support only two for enums

File: diktat-analysis.yml (hunk 1 above)
Comment (akuleshov7) [reply diktat-analysis.yml:14]: yeah, agreee

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt (hunk 2 above)
Comment (PhilipDukhov) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt]: That's done

**Review by akuleshov7 — APPROVED** [review]

**Pull request merged**